### DISCUSSION TITLE
Android questions (mainly a note saving issue)

### DISCUSSION THREAD

**Discussion by X1SDNA** [standardnotes/forum#2468] [General] [answered]
Hi - just some queries I have about behavior on the Android (v9) app , mainly with saving:

I have my notes on Windows 10, Linux & Android. At the moment I am hesitant to use Android. It seems to have a lot of trouble saving my notes. Even if I make only a small change to a note, it often just says 'Saving...' indefinitely. If I give up and move to Windows or Linux, there are conflict/duplicate issues.

If I restart my phone, then it will save the change that it couldn't save prior to restarting.

I also had Android issue whereby trashed notes kept appearing, but I signed out and signed in and that resolved it. Must've been a cache issue.

Also, can you please confirm what the folders extension looks like on Android? Is it just a list of tags, and not the dots with indented sub tags as per the desktop version? I just have a list of tags and wondering if that is normal. There appears to be no settings for Extensions on Android?

With the above behaviour, I'm not confident using it on Android for quick notes...I'd like to use it though. I can confirm that my internet connection is 100% fine, so regarding the long (or non-) saving issue, has this behaviour been seen before?

Thanks.

**Comment by atmoio** [comment]
Hmm that's strange. What version of the Android app?

**Comment by X1SDNA** [comment]
Hi - yes it's still happening. I can't find a more specific Android version number than '9' provided by my Samsung. Do you need the kernel version? Thanks.

**Comment by atmoio** [comment]
You can find the version of the app by opening the Settings screen, then scrolling down to the bottom. Some questions to better understand what may be affecting your saving:

1. Are you using a VPN? 
2. Which part of the world are you in?
3. Are you editing with the plain editor, or another editor?
4. Does this issue happen consistently, or intermittently?

**Comment by X1SDNA** [comment]
Thanks for your reply Mo. I'll monitor it for a bit longer and get back to you with more information. However, my Samsung only says version 9 for Android...I too was expecting something more specific. I'll update this soon.

**Comment by JaspalSuri** [comment]
Hi, I just want to briefly chime in here:

The version of Android that you're using is 9.X, but the version of Standard Notes that you have installed is a different number. Did you install it via the Play Store or F-Droid? As Mo said, you can view the version number of the app by navigating towards the bottom of the app's settings screen (which is separate from your phone's/OS's settings), which is accessible via the gear icon that appears in the left-sidebar. 

If you've downloaded it from the Play Store, **[please check here](https://play.google.com/store/apps/details?id=com.standardnotes&hl=en_US)** to see if it says "Open" or "Update" on your phone and press "Update" if needed. 

If you've downloaded it from F-Droid, you can get the latest version of the app (outside of the Play Store - feel free to ignore this part if you haven't) from GitHub, **[via this link](https://github.com/standardnotes/mobile/releases/download/3.0.21/app-release.apk)**.

**Comment by X1SDNA** [comment]
Sorry, I missed that...the version of the app itself is 3.0.22.

I'd like to monitor the situation and provide more specific info if I have further issues. Basically, the desktop versions sync and save like lightning, but for me, the Android version has moments where it struggles to save a note. Since I jump between the three (W10, Linux & Android), I hesitate to use Android so as to avoid conflict issues. However, I am only a few days into using Extended, and have a lot to learn, figure out, and get used to. Everyone's experience will be different depending on their usage environment/ecosystem, which I can understand makes your job challenging, especially for a note app with frequent and oscillating/choppy (and important) user behavior. (It also makes a good one more valuable).

I will try to provide more specifics after using it for a day or two.  I'll let you know. Thanks.

**Comment by X1SDNA** [comment]
To answer the other questions:

1. No VPN during use of the app.
2. Australia
3. Will have to pay attention to the editors in question and get back to you.
4. Yes, I'd say it is intermittent, however I can confirm that I always have a working internet connection, since I open a browser and test it when the issue arises.

**Comment by X1SDNA** [comment]
Update:

* The issue persists intermittently, however it seems to be a bigger problem when accessing a note for subsequent times in a session. That is, it generally saves fine for the first edit. If I close the app or go to other notes and come back, it is then that it often struggles to save.
* I usually just close it and open it a bit later, and it will usually save again, as if treating it like a 'first session', enough time evidently passing. However, if I leave my phone and app open as it is 'Saving...', it often will display the error:
```
The page at "https://extensions.standardnotes.org" says:
This extension is attempting to communicate with Standard Notes, but an error is preventing it from doing so. Please restart this extension and try again. OK.
``` 
Regarding this error, I press OK a gazillion times and then it works, or I close it and come back later.
* There are no internet connection issues for other things.
* This happens with Markdown Pro which I use the most, and also Plus Editor, but I haven't used Plus much, only in testing.
* Android version: `G950FXXUADTG6/G950FOXMADTG6/950FXXUADTG4`

**Comment by JaspalSuri** [comment]
Thanks for taking the time to provide that info! Hopefully we can reproduce this issue and find a fix.

**Comment by myreli** [comment, marked as answer]
Since the [new mobile experience](https://standardnotes.com/blog/our-new-mobile-experience-is-launching-soon), we haven't had similar reports, so I will mark it as resolved. Let us know if you find any problems.
